kolibri/core/analytics/measurements.py

```diff
@@ -0,0 +1,173 @@
+        recommended_time = '{:.2f} s'.format(requests.get(recommended_url).elapsed.total_seconds())
+        channels_url = format_url('/api/content/channel/?available=true', base_url)
+        channels_time = '{:.2f} s'.format(requests.get(channels_url).elapsed.total_seconds())
+
+    return (homepage_time, recommended_time, channels_time)
+
+
+def get_process_pid():
+    """
+    Returns the pid of the command in execution
+    :returns: Integer number with the pid of this command
+    """
+    this_command = psutil.Process()
+    return this_command.pid
+
+
+def get_machine_info():
+    """
+    Gets information on the memory, cpu and processes in the server
+    :returns: tuple of strings containing cpu percentage, used memory, free memory and number of active processes
+    """
+    used_cpu = str(psutil.cpu_percent())
+    used_memory = str(psutil.virtual_memory().used / pow(2, 20))  # In Megabytes
+    total_memory = str(psutil.virtual_memory().total / pow(2, 20))  # In Megabytes
+    total_processes = str(len(psutil.pids()))
+
+    return (used_cpu, used_memory, total_memory, total_processes)
+
+
+def get_kolibri_process_info():
+    """
+    Return information on the Kolibri process running in the machine
+    :returns: tuple of integers containing PID and TCP Port of
+              the running (if any) Kolibri server in this same machine
+    """
+    kolibri_pid = None
+    kolibri_port = None
+    try:
+        with open(PID_FILE, 'r') as f:
+            kolibri_pid = int(f.readline())
+            kolibri_port = int(f.readline())
+    except IOError:
+        pass  # Kolibri PID file does not exist
+    except ValueError:
+        pass  # corrupted Kolibri PID file
+    return (kolibri_pid, kolibri_port)
+
+
+def get_kolibri_process_cmd():
+    """
+    Retrieve from the OS the command line executed to run Kolibri server
+    :returns: tuple with command line and its arguments
+    """
+    kolibri_pid, _ = get_kolibri_process_info()
+    try:
+        kolibri_proc = psutil.Process(kolibri_pid)
+    except psutil.NoSuchProcess:
+        # Kolibri server is not running
+        raise KolibriNotRunning()
+    return kolibri_proc.cmdline()
+
+
+def get_kolibri_use(development=False):
+    """
+    Gets information on the memory and cpu usage of the current Kolibri process
+    :returns: tuple of strings containing cpu percentage and virtual memory used (in Kb)
+    """
+    kolibri_mem = kolibri_cpu = 'None'
+    kolibri_pid, _ = get_kolibri_process_info()
+
+    if kolibri_pid:
+        try:
+            kolibri_proc = psutil.Process(kolibri_pid)
+            kolibri_mem = str(kolibri_proc.memory_info().vms / pow(2, 10))
+            kolibri_cpu = str(kolibri_proc.cpu_percent())
+        except psutil.NoSuchProcess:
+            # Kolibri server is not running
+            raise KolibriNotRunning()
+
+    return (kolibri_cpu, kolibri_mem)
+
+
+class KolibriNotRunning(Exception):
```

Comment:
There's also `kolibri.utils.server.NotRunning` :)

Reply:
Thanks, I didn't know it. Fixing it.